hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -290,24 +314,45 @@ protected List<DatanodeDetails> chooseDatanodesInternal(
       unavailableNodes.addAll(excludedNodes);
     }
 
+    LOG.debug("Available racks excluding racks with used nodes: {}.", racks);
     if (racks.size() < additionalRacksRequired) {
       String reason = "Number of existing racks: " + racks.size()
               + "is less than additional required number of racks to choose: "
               + additionalRacksRequired + " do not match.";
       LOG.warn("Placement policy cannot choose the enough racks. {}"
                       + "Total number of Required Racks: {} Used Racks Count:" +
-                      " {}, Required Nodes count: {}",
+                      " {}, Required Nodes count: {}, fallback: {}.",
               reason, numberOfRacksRequired, usedRacksCntMap.size(),
-              nodesRequired);
-      throw new SCMException(reason,
-              FAILED_TO_FIND_SUITABLE_NODE);
+              nodesRequired, fallback);
+
+      /*
+      The number of racks which aren't already used by existing replicas
+      ("racks") and hence are available for new replicas are less than the
+      additional number of racks that replicas need to be on
+      ("additionalRacksRequired").
+      */
+      if (!fallback) {
```

Comment:
I don't really understand what you're suggesting. But the general idea here is that it's possible to fail some of the conditions above this, such as `nodesRequired < additionalRacksRequired`, and still get a new DN without having to increase `maxReplicasPerRack`.

Reply:
I meant we should recalculate maxReplicas per rack based on the number of available rack and not based on the topology configuration, since there could be cases some racks could be excluded completely.

Reply:
This should happen in the case racks.size() < additionalRacksRequired

Reply:
If you think this PR works as expected and what you're suggesting is an improvement that can be made separately, let's do that in https://issues.apache.org/jira/browse/HDDS-9054?